pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,218 @@
+			CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+		})
+		if err != nil {
+			glog.Errorf("Failed to list mount targets:%#v, %s", options, err)
+			return nil, err
+		}
+		if len(responseListMnt.Items) != 0 {
+			glog.Infof("Found mount targets to use")
+			rand.Seed(time.Now().Unix())
+			mntTargetSummary := responseListMnt.Items[rand.Int()%len(responseListMnt.Items)]
+			mntTargetResp = *getMountTargetFromID(ctx, *mntTargetSummary.Id, fileStorageClient)
+		} else {
+			// Mount target not created, create a new one
+			responseMnt, err := fileStorageClient.CreateMountTarget(ctx, filestorage.CreateMountTargetRequest{
+				CreateMountTargetDetails: filestorage.CreateMountTargetDetails{
+					AvailabilityDomain: availabilityDomain.Name,
+					SubnetId:           common.String(options.Parameters[subnetID]),
+					CompartmentId:      common.String(filesystem.client.CompartmentOCID()),
+					DisplayName:        common.String(fmt.Sprintf("%s%s", os.Getenv(volumePrefixEnvVarName), "mnt")),
+				},
+			})
+			if err != nil {
+				glog.Errorf("Failed to create a mount target:%#v, %s", options, err)
+				return nil, err
+			}
+			mntTargetResp = responseMnt.MountTarget
+		}
+	} else {
+		// Mount target already specified in the configuration file, find it in the list of mount targets
+		mntTargetResp = *getMountTargetFromID(ctx, options.Parameters[mntTargetID], fileStorageClient)
+	}
+
+	glog.Infof("Creating export set")
+	createExportResponse, err := fileStorageClient.CreateExport(ctx, filestorage.CreateExportRequest{
+		CreateExportDetails: filestorage.CreateExportDetails{
+			ExportSetId:  mntTargetResp.ExportSetId,
+			FileSystemId: response.FileSystem.Id,
+			Path:         common.String("/" + *response.FileSystem.Id),
+		},
+	})
+
+	if err != nil {
+		glog.Errorf("Failed to create export:%s", err)
+		return nil, err
+	}
+	serverIP := ""
+	if len(mntTargetResp.PrivateIpIds) != 0 {
+		privateIPID := mntTargetResp.PrivateIpIds[rand.Int()%len(mntTargetResp.PrivateIpIds)]
+		virtualNetworkClient := filesystem.client.VirtualNetwork()
+		getPrivateIPResponse, err := virtualNetworkClient.GetPrivateIp(ctx, core.GetPrivateIpRequest{
+			PrivateIpId: common.String(privateIPID),
+		})
+		if err != nil {
+			glog.Errorf("Failed to retrieve IP address for mount target:%s", err)
+			return nil, err
+		}
+		serverIP = *getPrivateIPResponse.PrivateIp.IpAddress
+	} else {
+		glog.Errorf("Failed to find server IDs associated with the mount target to provision a persistent volume")
+		return nil, fmt.Errorf("Failed to find server IDs associated with the mount target")
+	}
+
+	glog.Infof("Creating persistent volume on mount target with private IP address %s", serverIP)
+	return &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: *response.FileSystem.Id,
+			Annotations: map[string]string{
+				ociVolumeID: *response.FileSystem.Id,
+				ociExportID: *createExportResponse.Export.Id,
+			},
+			Labels: map[string]string{},
```

Comment:
Don't think this should be required

Reply:
Tested this and it does actually need to be initialised, as it is being used later on